codegen/gateway.go

```diff
@@ -210,25 +210,26 @@ func NewHTTPClientSpec(jsonFile string, clientConfigObj map[string]string, h *Pa
 
 	baseName := filepath.Base(filepath.Dir(jsonFile))
 
-	goFileName := filepath.Join(
-		h.CodeGenTargetPath(),
-		"clients",
-		baseName,
-		baseName+".go",
-	)
+	goFileName, err := h.GoFileGenPath(thriftFile, baseName+".go")
```

Comment:
This creates a deeply nested folder. we want to avoid that, the file name should be based on the `client-config.json` not the thrift file.